datadayessentials/config/_config.py

```diff
@@ -1,233 +1,161 @@
-        )
-        return LocalConfig.get_value_from_config(
-            ["azure", "storage_accounts", storage_account_ref]
-        )
+        Args:
+            variable_name (str): The name of the environment variable to retrieve.
 
-    @staticmethod
-    @catch_key_error
-    def get_data_lake():
-        data_lake= LocalConfig.get_value_from_config(
-            ["azure", "environments", LocalConfig.ENVIRONMENT, "data_lake"]
-        )
-        return data_lake
+        Returns:
+            str: The value of the requested environment variable.
 
-    @staticmethod
-    @catch_key_error
-    def get_environment():
-        return LocalConfig.get_value_from_config(
-            ["azure", "environments", LocalConfig.ENVIRONMENT]
-        )
+        Raises:
+            ValueError:If the specified environment variable is not found in local or cloud configuration after re-getting
+             variables.
+        """
+        if os.getenv(variable_name):
+            return os.getenv(variable_name)
+        else:
+            self.set_environment_variables()
+            if os.getenv(variable_name):
+                return os.getenv(variable_name)
+            else:
+                raise ValueError(
+                    f"Environment variable '{variable_name}' not found.\n"
+                    f"Make sure the variable is properly set in your cloud configuration."
+                )
 
-    @staticmethod
-    @catch_key_error
-    def get_environment_from_name(environment_name: str):
-        return LocalConfig.get_value_from_config(
-            ["azure", "environments", environment_name]
-        )
+    def set_environment_variables(self):
+        """
+        Sets environment variables based on the cloud configuration.
+        This method fetches values from Azure App Configuration and sets environment variables accordingly.
+        If in production, It will fetch values from Azure App Configuration and set environment variables accordingly.
+        If in development, it will fetch values from Azure App Configuration using manual authentication
+            and set environment variables accordingly.
+        """
+        if self.in_production_environment():
+            # TODO: Implement getting connection string in GitHub actions
+            self.create_azure_app_configuration_client_connection("placeholder")
+            self.set_environment_variables_from_cloud_config()
 
-    @staticmethod
-    @catch_key_error
-    def get_machine_learning_workspace():
-        machine_learning_workspace_ref = LocalConfig.get_value_from_config(
-            [
-                "azure",
-                "environments",
-                LocalConfig.ENVIRONMENT,
-                "machine_learning_workspace",
-            ]
-        )
-        return LocalConfig.get_value_from_config(
-            ["azure", "machine_learning_workspaces", machine_learning_workspace_ref]
-        )
+        if not self.in_production_environment():
```

Comment:
shouldnt need to check this status.  The if on line 89 checks for production and when the block finishes should exit the function.  If not in production, the rest of the code should run